lib/base64.nit

```diff
@@ -77,62 +77,74 @@ redef class NativeString
 
 	# Decodes `self` from base64
 	#
+	# `strict` is the reverse-equivalent of the `-i` option of
+	# the `base64` Unix command.
+	# i.e., it will check if all characters in `self` are from the
+	# `base64` alphabet and is correctly padded if needed.
+	# `strict` defaults to `true`
+	#
 	#      assert "c3RyaW5n".decode_base64 == "string"
 	#      assert "c3Rya\nW5n".decode_base64 == "string"
+	#      assert "c3RyaW5nCg==".decode_base64 == "string\n"
+	#      assert "c3RyaW5nCg".decode_base64(strict=false) == "string\n"
+	#      assert "c3RyaW5neQo=".decode_base64 == "stringy\n"
+	#      assert "c3RyaW5neQo".decode_base64(strict=false) == "stringy\n"
 	#
-	# REQUIRE: `length % 4 == 0`
-	private fun decode_base64(length: Int, padding: nullable Byte): Bytes do
-		if padding == null then padding = '='.ascii
+	private fun decode_base64(length: Int, padding: nullable Byte, strict: nullable Bool): Bytes do
```

Comment:
A sane recovering way might be to return a `nullable Bytes, nullable Error`, but this will have to wait for #839 I guess :)

Another sane way might be to just return a `nullable Bytes` without the `Error` part, but that would be cumbersome to understand why a `base64` could not be converted, although this can be an idea.

But honestly, if anyone has a better (usable, I'm not a fan of Javaesque contraptions, hence `Maybe[Bytes]` is not an option here due to its heaviness) idea to support this kind of behaviour without such a signature, I'm in !

Other than that, we're still on the debate of #1548, which is how do we handle errors in such a case.

I'd suggest that in the meantime, this should be left as-is (I can turn off the strict mode by default if you prefer), and we should talk about handling these kind of problems without putting too much burden on the end-user.